third_party/openthread/CMakeLists.txt

```diff
@@ -85,6 +85,10 @@ if (NOT OT_THREAD_VERSION STREQUAL "1.1")
     set(OT_LINK_METRICS_SUBJECT ON CACHE STRING "enable link metrics subject" FORCE)
 endif()
 
+if ((OT_BORDER_AGENT_EPSKC) AND (NOT OT_THREAD_VERSION STREQUAL "1.4"))
+    message(FATAL_ERROR "ePSKc requires Thread protocol version >= 1.4")
+endif()
```

Comment:
I wonder why it's necessary to enforce version 1.4 if we want to enable ePSKc feature.

Reply:
You can enable `OT_BORDER_AGENT_EPSKC` to enable ePSKc feature. 

The value here is just a default value which can be overriden:
https://github.com/openthread/openthread/blob/d0fbfb8c768eae85d6bf5099bd8129db1c0a94b1/src/core/config/border_agent.h#L84

Reply:
There was some discussion in https://github.com/openthread/ot-br-posix/pull/2259#discussion_r1591872385 regarding deprecating OTBR_ESPKC/OTBR_ENABLE_EPSKC, and https://github.com/openthread/openthread/pull/10503#issuecomment-2242492237 about the thread version + OT ephemeral key flag.

Thread 1.4 should have OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE, meanwhile it's hard to forbid enabling it for version less than 1.4.

For reference support, it seems all right to add Thread 1.4 build with `-DOT_THREAD_VERSION=1.4` specified Thoughts?